codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -39,22 +39,30 @@
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.Comparator;
 import java.util.HashMap;
 import java.util.HashSet;
-import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListSet;
 
 /**
  * Simple implementation of {@link Http2Connection}.
  */
 public class DefaultHttp2Connection implements Http2Connection {
 
+    private static final Comparator<Http2Stream> STREAMID_COMPARATOR = new Comparator<Http2Stream>() {
+        @Override
+        public int compare(Http2Stream o1, Http2Stream o2) {
+            return o1.id() - o2.id();
```

Comment:
@nmittler @Scottmitch I believe that this `Comparator` should actually order the streams based on their priority. I am not too familiar with the implementation of priorities in Netty, but having glanced over the code it seems to be quite an expensive operation to do (check for parent, check for child and so on). Also according to my understanding of HTTP2 Stream priority no total order between the streams is established, which means we would have to define the order of some of the edge cases ourselves? Also prior to this change the `LinkedHashSet` doesn't order the streams by priority either. So this change basically just preserves the current behavior. WDYT?